augurpy/estimators.py

```diff
@@ -1,25 +1,57 @@
-from typing import Dict, List, Optional, Union
+from typing import Dict, Literal, Union
 
 from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
 from sklearn.linear_model import LogisticRegression
+from typing_extensions import TypedDict
+
+
+class Params(TypedDict, total=False):
+    """Type signature for random forest and logistic regression parameters."""
+
+    n_estimators: int
+    max_depth: int
+    max_features: Union[Literal["auto"], Literal["log2"], Literal["sqrt"], int, float]
+    penalty: Union[Literal["l1"], Literal["l2"], Literal["elasticnet"], Literal["none"]]
 
 
 def create_estimator(
-    classifier: Union[RandomForestClassifier, RandomForestRegressor, LogisticRegression],
-    rf_params: Optional[List],
-    lr_params: Optional[List],
+    classifier: Union[Literal["rf"], Literal["lr"]],
+    params: Params,
+    classification: bool = True,
 ) -> Union[RandomForestClassifier, RandomForestRegressor, LogisticRegression]:
     """Estimator object.
 
     Args:
         classifier: classifier to use in calculating the area under the curve either random forest or logistic regression
-        rf_params: list of parameters for random forest
-        lr_params: list of parameters for logistic regression
+        params: list of parameters for estimator
 
     Returns:
         Estimator object that classifies.
     """
-    pass
+    if classifier == "rf" and classification:
+        estimator = RandomForestClassifier(
+            n_estimators=params.get("n_estimators", 100),
+            max_depth=params.get("max_depth", None),
+            max_features=params.get("max_features", "auto"),
+        )
+        print(estimator, " Object created.")
+
+    elif classifier == "rf" and not classification:
+        estimator = RandomForestRegressor(
+            n_estimators=params.get("n_estimators", 100),
+            max_depth=params.get("max_depth", None),
+            max_features=params.get("max_features", "auto"),
+        )
+        print(estimator, " Object created.")
+
+    elif classifier == "lf" and not classification:
+        estimator = LogisticRegression()
+        print(estimator + " Object created.")
+
+    else:
+        raise Exception("Missing valid input.")
```

Comment:
Always try to offer a resolution. "Missing valid input. Valid input is xyz".

Reply:
done